addons/metadata.integral.artists/lib/musicbrainz.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: UTF-8 -*-
+def musicbrainz_artistfind(data):
+    artists = []
+    for item in data.get('artists',[]):
+        artistdata = {}
+        artistdata['artist'] = item['name']
+        artistdata['thumb'] = ''
+        artistdata['genre'] = ''
+        artistdata['born'] = ''
+        artistdata['mbid'] = item['id']
+        if item.get('score',1):
+            artistdata['relevance'] = str(item['score'] / 100.00)
+        artists.append(artistdata)
+    return artists
+
+def musicbrainz_artistdetails(data):
+    artistdata = {}
+    artistdata['artist'] = data['name']
+    artistdata['mbartistid'] = data['id']
+    if data.get('life-span','') and data.get('type',''):
+        begin = data['life-span'].get('begin', '')
+        end = data['life-span'].get('end', '')
+        if data['type'] == 'Group':
+            artistdata['formed'] = begin
```

Comment:
From Musicbrainz documentation https://musicbrainz.org/doc/Artist artist type  can be "Person", "Group", "Orchestra", "Choir", "Character", or "Other" although I would not like to guarentee they stick to that.  Types "Orchestra" and "Choir" need to be treated like "Group" when it comes to having "formed" date rather than "born" etc. 

Also I'm not sure if every artist has a type, and if it doesn't but still has begin or end data then it would still be good to store it somewhere

Reply:
spot on. will fix that :-)

yes it's possible an artist does not have a type, but storing metadata 'somewhere' is currently not supported by kodi ;-)

Reply:
How about if type = "" or something unexpected then save "begin" and "end" from Musicbrainz to "Born" and "Died" as if it was a person rather than discarding the data.